ledger/eval.go

```diff
@@ -624,19 +623,15 @@ func (eval *BlockEvaluator) prepareEvalParams(txgroup []transactions.SignedTxn)
 			continue
 		}
 
-		// Initialize group without ApplyData lazily
-		if groupNoAD == nil {
-			groupNoAD = make([]transactions.SignedTxn, len(txgroup))
-			for j := range txgroup {
-				groupNoAD[j] = txgroup[j]
-			}
-			minTealVersion = logic.ComputeMinTealVersion(groupNoAD)
+		// Initialize minTealVersion lazily
+		if minTealVersion == 0 {
```

Comment:
I think that you meant `if i == 0`, right ?

Reply:
No because I'm trying to follow the original logic of computing this value lazily, ie only if we encounter a txn of type `protocol.ApplicationCallTx`. However, checking if the value is 0 might not be the right thing to do here, perhaps introduce another variable. Thoughts?

Reply:
yes, you're correct about this one, but comparing minTealVersion to 0 isn't the right solution either since a zero is a legit return code from logic.ComputeMinTealVersion.
given that the entire purpose of this loop ( now ) is to calculate the minTealVersion, I would suggest dropping the if statement completely and break right after calling ` logic.ComputeMinTealVersion`. Does that sounds like a good solution?

Reply:
I would suggest creating another loop before this one, to find calculate the value of the minTealVersion and break once it's calculated.